plugins/woocommerce/src/Blocks/BlockTypes/ProductCollection.php

```diff
@@ -531,6 +538,29 @@ private function get_final_query_args( $common_query_values, $query, $is_exclude
 		return $result;
 	}
 
+	/**
+	 * Get query args for preview mode. These query args will be used with WP_Query to fetch the products.
+	 *
+	 * @param array           $args    Query args.
+	 * @param WP_REST_Request $request Request.
+	 */
+	private function get_preview_query_args( $args, $request ) {
+		$collection_query = array();
+
+		/**
+		 * In future, Here we will modify the preview query based on the collection name. For example:
+		 *
+		 * $product_collection_query_context = $request->get_param( 'productCollectionQueryContext' );
+		 * $collection_name                  = $product_collection_query_context['collection'] ?? '';
+		 * if ( 'woocommerce/product-collection/on-sale' === $collection_name ) {
+		 *      $collection_query = $this->get_on_sale_products_query( true );
+		 * }.
+		 */
+
+		$args = $this->merge_queries( $args, $collection_query );
+		return $args;
+	}
+
```

Comment:
I can't think of a good reason we should exclude e2e tests, please include some for this.

Reply:
Added E2E tests in [4bcec0b](https://github.com/woocommerce/woocommerce/pull/46369/commits/4bcec0b8eb20940bc0b5a8eaafee912d8357ba48) 🤝